src/canvas/helper.ts

```diff
@@ -46,12 +51,17 @@ export function createRadialGradient(
     let x = obj.x == null ? 0.5 : obj.x;
     let y = obj.y == null ? 0.5 : obj.y;
     let r = obj.r == null ? 0.5 : obj.r;
+
     if (!obj.global) {
         x = x * width + rect.x;
         y = y * height + rect.y;
         r = r * min;
     }
 
+    x = isSafeNum(x) ? x : 0.5;
+    y = isSafeNum(y) ? y : 0.5;
+    r = isSafeNum(r, true) ? r : 0.5;
```

Comment:
Seems only one number needs to be check positive. In this case I will prefer keeping the `isSafeNum` simpler and more cohesion that only accept one parameter `number`. Monomorphic function is always better than polymorphic function, on both code style and performance.

So the `r` check can be
```ts
r = isSafeNum(r) && r >= 0 ? r : 0.5
```